### PR TITLE
Fix pylint's pylint-redefined-outer-name

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: h/cli/commands/shell.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=redefined-outer-name
```
Comment (marcospri) [inline h/cli/commands/shell.py:1]: There's a few if this `commands` files that pylint doesn't love

File: h/services/group_create.py [hunk 2]
```diff
@@ -10,15 +10,15 @@
 
 
 class GroupCreateService:
-    def __init__(self, session, user_fetcher, publish):
+    def __init__(self, db, user_fetcher, publish):
```
Comment (marcospri) [inline h/services/group_create.py:13]: Will do another PR to add `db` as a name we like

File: h/services/user_signup.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from h.emails import signup
 from h.models import Activation, Subscriptions, User, UserIdentity
 from h.services.exceptions import ConflictError
-from h.tasks import mailer
+from h.tasks import mailer as tasks_mailer
```
Comment (marcospri) [inline h/services/user_signup.py:9]: I  think this is better than taking a mailer_ kwarg

File: h/cli/commands/authclient.py [hunk 4]
```diff
@@ -30,19 +30,19 @@ def add(ctx, name, authority, type_):
     """
     request = ctx.obj["bootstrap"]()
 
-    authclient = models.AuthClient(name=name, authority=authority)
+    authclient_ = models.AuthClient(name=name, authority=authority)
```
Comment (jon-betts) [inline h/cli/commands/authclient.py:33]: This is a little cludgy to me. I think we could go with `auth_client` or just `client` as it's pretty clear which kind we have by context.

The other way to go is to call `def authclient` something like `authclient_command` or `authclient_group`

File: h/views/api/annotations.py [hunk 5]
```diff
@@ -47,8 +47,8 @@ def search(request):
 
     separate_replies = params.pop("_separate_replies", False)
 
-    search = search_lib.Search(request, separate_replies=separate_replies)
-    result = search.run(params)
+    search_ = search_lib.Search(request, separate_replies=separate_replies)
+    result = search_.run(params)
```
Comment (jon-betts) [inline h/views/api/annotations.py:51]:
```suggestion
    result = search_lib.Search(request, separate_replies=separate_replies).run(params)
```

**Review by jon-betts — APPROVED** [review]
Honest to god, none of the following feedback matters, so feel free to merge, but I think this PR takes quite a direct approach to fixing the issue on the line it's reported, where there might be some other ways of doing it.

Some suggestions in a random order:

* The redeclarations are where the issue is flagged, but sometimes it's the first declaration that might make more sense to rename
* In particular our function names are very noun-y rather than verb-y in a few places
* We often use `db_session` in our classes which would avoid some of the hits
* There are quite a few places where no variable is required at all
* I prefer not adding `_` to the end if we can think of another more descriptive word instead that works just as well

File: h/session.py [hunk 6]
```diff
@@ -76,15 +76,15 @@ def _current_groups(request, authority):
 
 
 def _group_model(route_url, group):
-    model = {"name": group.name, "id": group.pubid, "public": group.is_public}
```
[the remark this answers is not in the document]
Comment (marcospri) [reply h/session.py:79]: There already group as a pos argument.

File: h/cli/commands/devdata.py [hunk 7]
```diff
@@ -21,10 +21,10 @@
 class DevDataFactory:
     """Factory class for inserting standard dev data into the DB."""
 
-    def __init__(self, request, devdata):
+    def __init__(self, request, devdata_):
         self.db = request.db
         self.tm = request.tm
-        self.devdata = devdata
+        self.devdata = devdata_
```
[the remark this answers is not in the document]
Comment (marcospri) [reply h/cli/commands/devdata.py:27]: I think that will change the name of the command itself

[the remark this answers is not in the document]
Comment (jon-betts) [reply h/cli/commands/devdata.py:27]: Oh.. fair enough

File: h/session.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jon-betts) [reply h/session.py:79]: :facepalm:

**Review by jon-betts — APPROVED** [review]
